src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -9038,23 +9038,15 @@ public class MyAttribute : System.Attribute
 }
 ";
             var comp = CreateCompilation(source);
+
+            // PROTOTYPE: Should report CS0103: The name 'parameter' does not exist in the current context
```

Comment:
I didn't understand why the diagnostic is gone. How is this parsed now?

Reply:
it previously just went off the rails (as the `[...] delegate` form isn't legal at all).  Now, it gets parsed as `System.Func<int, int> x = [My(nameof(parameter))]` (i.e. a local varaible initialized to a legal collection literal).  then we folllow with a missing semicolon.  then we have a random expr-statement containing the anonymous-method, which itself does not have a semicolon.

Reply:
Adding parsing test to show this.

Reply:
Test added.